tsdb/head_append.go

```diff
@@ -1968,6 +1980,7 @@ func (a *headAppender) Rollback() (err error) {
 		series.pendingCommit = false
 		series.Unlock()
 	}
+	a.unmarkCreatedSeriesAsPendingCommit()
```

Comment:
I can’t immediately see why this is inconsistent with the loops in `defer` a few lines earlier.

Reply:
I tried to be consistent with the fact that we do this after processing `a.samples` which is a requirement in `Commit`. 

It is true that at the end of this method we do `a.log()` and write created series, so maybe it's worth keeping them until then? But also we just removed `pendingCommit` from most of them (from the ones that we created with samples) in the lines above. :-/

As you mentioned offline, a bool doesn't really work for `pendingCommit`, and IMO it should be a reference counter, but I would like to send that change separately as it will likely raise more questions.

Reply:
I moved a call into a deferred one from above, so it would happen after writing WAL, consistent with the one in the log.